### PR TITLE
[IMP] web: add support for new granularities in mock server

### PR DESCRIPTION
**Pull request by pro-odoo**
This PR adapt the Hoot and legacy mock servers to 7ce55ca15cdee848f4f10eb67f92703335d42070

Task: 3899610

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/tests/_framework/mock_server/mock_model.js [hunk 1]
```diff
@@ -260,16 +287,36 @@ const formatFieldValue = (fields, groupByField, val) => {
         switch (aggregateFunction) {
             // The year is added to the format because is needed to correctly compute the
             // domain and the range (startDate and endDate).
```
Comment (aab-odoo) [inline addons/web/static/tests/_framework/mock_server/mock_model.js:289]: this comment applies to the "hour" case I think

File: addons/web/static/tests/_framework/mock_server/mock_model.js [hunk 2]
```diff
@@ -260,16 +287,36 @@ const formatFieldValue = (fields, groupByField, val) => {
...
+            case "month_number":
+                return date.month;
             case "quarter":
                 return `Q${date.toFormat("q yyyy")}`;
+            case "quarter_number":
+                return date.quarter;
             case "year":
                 return date.toFormat("yyyy");
+            case "year_number":
+                return date.year;
```
Comment (aab-odoo) [inline addons/web/static/tests/_framework/mock_server/mock_model.js:319]: there's a lot of duplicated code here, maybe try to factorize it? Looks like second/minute/hour is obviously for datetime only, but all the rest is the same for both date and datetime

File: addons/web/static/tests/_framework/mock_server/mock_model.js [hunk 3]
```diff
@@ -1921,52 +1995,61 @@ export class Model extends Array {
...
-                                startDate = parseDateTime(value, { format: "y" });
-                                endDate = startDate.plus({ years: 1 });
-                                break;
-                            }
-                            case "month":
-                            default: {
-                                startDate = parseDateTime(value, { format: "MMMM yyyy" });
-                                endDate = startDate.plus({ months: 1 });
-                                break;
+                        if (!READ_GROUP_NUMBER_GRANULARITY.includes(dateRange)) {
```
Comment (aab-odoo) [inline addons/web/static/tests/_framework/mock_server/mock_model.js:1998]: maybe rename dateRange into granularity

File: addons/web/static/tests/_framework/mock_server/mock_model.js [hunk 4]
```diff
@@ -1921,52 +1995,61 @@ export class Model extends Array {
...
+                            const to = serialize(endDate);
+                            group.__range[gbField] = { from, to };
+                            group.__domain = [
+                                [fieldName, ">=", from],
+                                [fieldName, "<", to],
+                                ...group.__domain,
+                            ];
+                        } else {
+                            group.__domain = [
+                                [`${fieldName}.${dateRange}`, "=", parseInt(value, 10)],
```
Comment (aab-odoo) [inline addons/web/static/tests/_framework/mock_server/mock_model.js:2049]: isn't value already a number?

File: addons/web/static/tests/mock_server/mock_server.test.js [hunk 5]
```diff
@@ -500,6 +500,66 @@ test("performRPC: read_group, group by date", async () => {
     ]);
 });
 
+test("performRPC: read_group, group by date with number granularity", async () => {
+    await makeMockServer();
+
+    const all_granularity = [
```
Comment (aab-odoo) [inline addons/web/static/tests/mock_server/mock_server.test.js:506]: camelcase

File: addons/web/static/tests/_framework/mock_server/mock_model.js [hunk 6]
```diff
@@ -243,36 +280,14 @@ const formatFieldValue = (fields, groupByField, val) => {
...
-                return `W${date.toFormat("WW kkkk")}`;
-            case "quarter":
-                return `Q${date.toFormat("q yyyy")}`;
-            case "year":
-                return date.toFormat("yyyy");
-            default:
-                return date.toFormat("MMMM yyyy");
-        }
+        return aggregateFunction in DATETIME_FORMAT
+            ? DATE_FORMAT[aggregateFunction](date)
```
Comment (aab-odoo) [inline addons/web/static/tests/_framework/mock_server/mock_model.js:289]: DATETIME_FORMAT

**Comment by pro-odoo** [comment]
@aab-odoo Updated :)

**Review by aab-odoo — APPROVED** [review]
Thanks @pro-odoo 

robodoo rebase-ff r+

**Pull request closed without merging**